pysilcam/acquisition.py

```diff
@@ -97,30 +106,167 @@ def print_camera_config(camera):
     config_info = '\n'.join(['{0}: {1}'.format(a, camera.getattr(a))
                              for a, b in config_info_map])
 
-    logger.info(config_info) # TODO: Should this be printed?
+    logger.info(config_info)  # TODO: Should this be printed?
     logger.debug(config_info)
 
 
+def silcam_load(filename):
+    #Load the raw image from disc depending on filetype
+    if filename.endswith('.silc'):
+        img0 = np.load(filename, allow_pickle=False)
+    elif filename.endswith('.silc_mono'):
+        
+        img0 = np.load(filename, allow_pickle=False)
+    else:
+        img0 = imageio.imread(filename)
+    
+    # this is a quick fix to load mono 8 bit images
+    if len(np.shape(img0)) == 2:
+        img_mono =  img0
+        r, c = np.shape(img_mono)
+        img0 = np.zeros([r, c, 3], dtype=np.uint8)
+        img0[:, :, 0] = img_mono
+        img0[:, :, 1] = img_mono
+        img0[:, :, 2] = img_mono
+    
+    return img0
+
+
+def silcam_name2time(fname):
+    timestamp = pd.to_datetime(os.path.splitext(fname)[0][1:])
+    return timestamp
+
+
 class Acquire():
     '''
     Class used to acquire images from camera or disc
     '''
-    def __init__(self, USE_PYMBA=False, FAKE_PYMBA_OFFSET=0):
+    def __init__(self, USE_PYMBA=False, datapath=None, writeToDisk=False,
+                 gui=None, raw_image_queue=None,
+                 max_n_images=None, start_image_offset=0):
+        
+        self.gui = gui
+        self.raw_image_queue = raw_image_queue
+        self.datapath = datapath
+        self.max_n_images = max_n_images
+        self.start_image_offset = int(start_image_offset)
+
         if USE_PYMBA:
-            self.pymba = pymba
-            self.pymba.get_time_stamp = lambda x: pd.Timestamp.now()
-            logger.info('Pymba imported')
-            self.get_generator = self.get_generator_camera
+            self.vimba = Vimba
+            logger.info('Vimba imported')
+
+            if datapath != None:
```

Comment:
E711: comparison to None should be 'if cond is not None:'  <!-- Automatically posted by Lintly -->